src/vector/index.ts

```diff
@@ -79,6 +79,10 @@ function checkBrowserFeatures(): boolean {
     // and older Firefox has the former but not the latter, so we add our own.
     window.Modernizr.addTest("intlsegmenter", () => typeof window.Intl?.Segmenter === "function");
 
+    // Basic test for WebAssembly support. We could also try instantiating a simple module,
+    // although this would start to make (more) assumptions about how rust-crypto loads its wasm.
+    window.Modernizr.addTest("wasm", () => typeof WebAssembly === "object" && typeof WebAssembly.Module === "function");
```

Comment:
I don't think this is sufficient, I believe rust crypto needs the modern API for loading it also. e.g. https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/instantiateStreaming_static rather than https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/instantiate_static (FF 58 vs FF 52)

Reply:
So it actually doesn't, because it seems the lowest-common-denominator way was the only reliable way of loading it portably, so it actually uses `WebAssembly.instantiate()` in practice. We could certainly try loading a simple module using that: that would cover CSP issues too, but we'd be assuming more details of how the library works, which could change without us knowing so we could end up saying it won't work even when it actually would.